test/testmain.cpp

```diff
@@ -102,3 +102,26 @@ TEST_CASE("flex parser") {
   REQUIRE(token.line == 1);
   REQUIRE(token.lexeme == test_string);
 }
+
+TEST_CASE("flex parser_2") {
+  SymbolTable sym;
+  std::string test_string = "a = 5;";
+  std::stringstream sf(test_string);
+
+  std::vector<std::tuple<std::string, int, Tokentype>> expected_output = {
+    std::make_tuple("a", 1, Tokentype::Identifier),
+    std::make_tuple("=", 1, Tokentype::OpAssign),
+    std::make_tuple("5", 1, Tokentype::Number),
+    std::make_tuple(";", 1, Tokentype::ptSemicolon)
+  };
+
+  Lexer* lexer;
+  lexer = new FLexer(sf, sym);
+  Token token;
+  for (auto test_tuple : expected_output) {
+    lexer->get_next(token);
+    REQUIRE(token.lexeme == std::get<0>(test_tuple));
+    REQUIRE(token.line == std::get<1>(test_tuple));
+    REQUIRE(token.type == std::get<2>(test_tuple));
+  }
```

Comment:
Can you add the line "delete lexer;" at the end in this and the other test I did?

I forgot adding it back then and just realised here 😅 . Nothing should happen in this case but it's a memory leak anyways.

Reply:
Fine now?

Reply:
:shipit: